src/swaps/jupiter/JupiterSwapProvider.ts

```diff
@@ -0,0 +1,243 @@
+          };
+        },
+      },
+      FAILED: {
+        step: 1,
+        label: 'Swap Failed',
+        filterStatus: 'REFUNDED',
+        notification() {
+          return { message: 'Swap failed' };
+        },
+      },
+    };
+  }
+  protected _txTypes(): Record<string, string | null> {
+    return {
+      SWAP: 'SWAP',
+    };
+  }
+  protected _fromTxType(): string | null {
+    return this._txTypes().SWAP;
+  }
+  protected _toTxType(): string | null {
+    return null;
+  }
+  protected _totalSteps(): number {
+    return 2;
+  }
+  protected _timelineDiagramSteps(): string[] {
+    return ['SWAP'];
+  }
+
+  public async getSupportedPairs() {
+    return [];
+  }
+
+  public async getQuote(quoteRequest: QuoteRequest) {
+    const { from, to, amount } = quoteRequest;
+    const fromInfo = cryptoassets[from];
+    const toInfo = cryptoassets[to];
+
+    // only for Solana network swaps
+    if (fromInfo.chain !== ChainId.Solana || toInfo.chain !== ChainId.Solana || amount.lt(0)) {
+      return null;
+    }
+
+    const solMintAddress = 'So11111111111111111111111111111111111111112';
+
+    const fromAddress = fromInfo.contractAddress || solMintAddress;
+    const toAddress = toInfo.contractAddress || solMintAddress;
+
+    const fromAmountInUnit = currencyToUnit(fromInfo, new BN(amount)).toFixed();
+
+    const { data } = await axios.get(
+      `https://quote-api.jup.ag/v1/quote?inputMint=${fromAddress}&outputMint=${toAddress}&amount=${fromAmountInUnit}&slippage=0.5&`
+    );
+
+    if (!data.data?.[0]) {
+      return null;
+    }
+
+    const info = data.data[0];
+
+    return {
+      from,
+      to,
+      fromAmount: fromAmountInUnit.toString(),
+      toAmount: info.outAmount.toString(),
+      info: {
+        ...info,
+        toAddress,
+      },
+    };
+  }
+
+  public async newSwap(quoteInput: SwapRequest<JupiterSwapHistoryItem>) {
+    const connection = new Connection(
+      'https://solana--mainnet.datahub.figment.io/apikey/d7d9844ccf72ad4fef9bc5caaa957a50',
+      'confirmed'
+    );
+    const { network, walletId, quote } = quoteInput;
+    const client = this.getClient(network, walletId, quote.from, quote.fromAccountId);
+    const [{ address }] = await client.wallet.getAddresses();
+
+    const transactions = await this._getTransactions(quote.info, address);
+
+    let swapTx;
+    const txTypes = Object.keys(transactions);
+
+    /* Full swap flow might be constructed from 3 different transactions:
+      1. Setup transaction  - used to handle creating ATA or Serum open order accounts
+      2. Swap transaction  - performing the actual swaps
+      3. Clean up transaction - unwrap SOL if a SOL swap.
+      You must send each transaction sequentially in the order:
+      setupTransaction -> swapTransaction -> cleanupTransaction and wait for each to be 'confirmed' before sending the next one.
+    */
+
+    for (let i = 0; i < txTypes.length; i++) {
```

Comment:
Because the transactions are really fast, they don't need to be in different steps?

Reply:
They are in different steps:

`You must send each transaction sequentially in the order:
      setupTransaction -> swapTransaction -> cleanupTransaction and wait for each to be 'confirmed' before sending the next one.`